server/src/test/java/org/apache/druid/client/DirectDruidClientTest.java

```diff
@@ -349,4 +349,77 @@ public void testQueryInterruptionExceptionLogMessage()
     Assert.assertEquals(hostName, actualException.getHost());
     EasyMock.verify(httpClient);
   }
+
+  @Test
+  public void testQueryTimeoutFromFuture()
```

Comment:
Is it simpler to write a direct unit test for JsonParserIterator to cover all the various scenarios?

Reply:
The only thing that can use the functionality introduced in this PR is `DirectDruidClient` so it doesn't seem unreasonable to me to be here. 

From my perspective this was significantly _simpler for me_, since I just had to copy-pasta another test and adjust behavior to hit what I want. There isn't a  `JsonParserIteratorTest` so creating that from scratch setting up everything there to make sure it does all of its things seemed like a lot more work to the me of last night.

From a completeness perspective it is probably _better_, but I am also lazy and have a bunch of other stuff I would rather be doing, so would prefer to leave this as is for now instead of write tests for _all_ of `JsonParserIterator` unless pushed hard on this.

This file used to be in `DirectDruidClient` before it was moved out to its own file so it could be used by `SystemSchema` and `MetadataSegmentView`, which is probably why it doesn't have a test of its own currently.

Reply:
ah, i moved this file out , agree a separate test would be good, will try to add one soon(?)